### ISSUE TITLE
Handle bayer for color cameras

### ISSUE THREAD

**Issue by fft001** [vidi-systems/ViDi.Net.BaslerCamera#1]

**Comment by SimenDeath** [comment]
We would also appreciate handeling bayer format

**Comment by fft001** [comment]
Hi SimenDeath,

Do you have a Basler color camera ? I don't have one so I cannot test directly the code.
can you try out the code in the branch colorsupport ?

Thank you !

**Comment by SimenDeath** [comment]
Yes I will test it for you today ! :) thanks for fast response .. also i think problem is that widi will use 3 layers as input instead of one - current solution for me was use one layer and change camera settings to Mono8 grayscale.

Jan Šimon

**Comment by SimenDeath** [comment]
So i tried new branch with Mono8 and also with Bayer (grayscale or color settings with Bastler camera) and i am getting error: "To read the SFNC version from the camera device, the connection to the camera must be open."

I was also trying to make changes by myself. I was not able to find PLCamera.PixelCoding attribute and also my camera support only BayerBG8, BayerBG12, BayerBG12p and YCbCr 422 format.

**Comment by SimenDeath** [comment]
So i programmed some changes but i am not able to create pull request :) will you be able to give me rights to create pull requests ?

**Comment by fft001** [comment]
Done., you should have the rights.

Sorry I did a mistake in the branch, the set settings should occur after open ideed.

**Comment by SimenDeath** [comment]
Thanks. I am kind of new in Git can you please check if the commit and code is here ? :)

PS: good think is that now it can handle every format because its using Bitmap. Problem is that Preview in control panel and sended image for recognition is from same source. Bitmap got locked data so you can do one think at the time. I got solution right now that i am not using Preview in ControlPanel plugin and everythink works great :).
